### PR TITLE
UnitControl: Use `hideHTMLArrows` prop to hide spin buttons

### PR DESCRIPTION
**Pull request by mirka**
## What?

Uses NumberControl's `hideHTMLArrows` prop to hide the spin buttons, instead of custom CSS.

## Why?

While going through the NumberControl types (#43791), I noticed that UnitControl was adding its own CSS styles to hide the spin buttons, when in fact NumberControl provided that option via the `hideHTMLArrows` prop.

## How?

Remove the CSS overrides and use the designated prop.

## Testing Instructions

1. `npm run storybook:dev` and see the UnitControl story.
2. Spin buttons should be hidden, even when `disableUnit` is true. Check Chrome/Firefox/Safari.

### CODE REVIEW

File: packages/components/src/box-control/unit-control.js [hunk 1]
```diff
@@ -35,7 +35,6 @@ export default function BoxUnitControl( {
 				<UnitControl
 					aria-label={ label }
 					className="component-box-control__unit-control"
-					hideHTMLArrows
```
Comment (mirka) [inline packages/components/src/box-control/unit-control.js:38]: In BoxControl, spin buttons were always hidden, even when units were disabled. (Units get disabled in the "linked" control when the units of the separate box sides are "mixed".)

However, other BoxControl-like controls in the block editor like the [Spacing](https://github.com/WordPress/gutenberg/blob/ca37b82de4700065d5fedbe8829768144e8bfaeb/packages/block-editor/src/components/spacing-sizes-control/spacing-input-control.js#L219) and [Border Radius](https://github.com/WordPress/gutenberg/blob/ca37b82de4700065d5fedbe8829768144e8bfaeb/packages/block-editor/src/components/border-radius-control/all-input-control.js#L58) controls use a raw UnitControl and do not hide spin buttons when units are disabled.

@jasmussen Any opinions on whether we should unify these hiding behaviors in the UnitControls? The options are:

1. In a UnitControl, always show spin buttons when the units are disabled.
2. In a UnitControl, always hide spin buttons, regardless of whether units are disabled.
3. The consumers of UnitControl should be able to choose whether spin button are hidden. (In this case, we'll also want to iron out the inconsistencies in the BoxControl-like controls I described above.)

Comment (jasmussen) [reply packages/components/src/box-control/unit-control.js:38]: Just to be sure I understand you right, when you refer to spin buttons, you refer to these?

image

If yes, then I haven't seen a case where we need to/should surface these in the editor. We have a click & drag behavior:
click and drag

There's also up arrow for incrementing, down arrow for decrementing, and shift up/down for doing that in increments of ten. Those may be less discoverable, but feel far more ergonomic to use than the tiny little hit areas of the up/down arrows. What do you think?

Comment (ciampo) [reply packages/components/src/box-control/unit-control.js:38]: IMO, in general showing spin buttons would be a good idea and would improve the input's usability. Spin buttons would offer a very "visually explicit" alternative to increase/decrease the value in the input field (while every other alternative mentioned is more implicit and hard to discover).


> Any opinions on whether we should unify these hiding behaviors in the UnitControls? The options are:
> 
> 1. In a UnitControl, always show spin buttons when the units are disabled.
> 2. In a UnitControl, always hide spin buttons, regardless of whether units are disabled.
> 3. The consumers of UnitControl should be able to choose whether spin button are hidden. (In this case, we'll also want to iron out the inconsistencies in the BoxControl-like controls I described above.)

I would vote for 1 or 3.

> Units get disabled in the "linked" control when the units of the separate box sides are "mixed"

Just to understand — in that case, `BoxControl` seems to show the word "Mixed", and so I'm not sure how spin buttons would be useful in this scenario?

Screenshot 2022-09-09 at 18 23 06

Comment (jasmussen) [reply packages/components/src/box-control/unit-control.js:38]: Given how small the buttons are, and the fact that you can at any one point type in an explicit value, the spin buttons really don't seem worth optimizing for.

Comment (mirka) [reply packages/components/src/box-control/unit-control.js:38]: My original intent was to discuss step buttons in UnitControl only, but I guess now we're discussing whether to nix step buttons in general, including normal NumberControls.

I just remembered that we have a huge move in the other direction, waiting on our todo list to implement:

NumberControl mockup with large step buttons

And IIRC, @mtias wrote somewhere that UnitControl was originally intended to be accompanied by a slider, like so:

UnitControl with slider

So perhaps the unifying idea is that number inputs *should* have some sort of visually obvious auxiliary control, that is *usable on a touch device*. In this case, the tiny built-in step buttons do not satisfy this requirement anyway.

tl;dr — I guess I'm fine with removing the step buttons from UnitControl right now. And they'll also be removed from NumberControl once the large custom step buttons are implemented.

Comment (jasmussen) [reply packages/components/src/box-control/unit-control.js:38]: The slider and the bigger plus/minuses are both more ergonomic tools, excellent point.

Comment (mtias) [reply packages/components/src/box-control/unit-control.js:38]: Indeed, the big + – is meant to cover incremental steps where each step is meaningful, like line height, while the slider is meant to cover wider ranges. A + – input should not be paired with a slider; custom font-size really need to be paired with the slider.

Comment (ciampo) [reply packages/components/src/box-control/unit-control.js:38]: In case where `UnitControl` will need to be adjusted with small incremental steps, would we show the big "+" / "-" buttons next to the unit dropdown?

Comment (mtias) [reply packages/components/src/box-control/unit-control.js:38]: What's the unit dropdown? + / – should show inside the control like the line height example.

Comment (ciampo) [reply packages/components/src/box-control/unit-control.js:38]: > What's the unit dropdown?

The unit dropdown in the unit control component is the dropdown that is shown when clicking on the unit (e.g. "px"):

Screenshot 2022-09-13 at 12 15 12
Screenshot 2022-09-13 at 12 15 05

> + / – should show inside the control like the line height example.

The line height control doesn't show a unit, and so it would be using a `NumberControl`.

My question was about instances in which a `UnitControl` is used (which means that the unit dropdown is shown), as was also pointed out by @mirka at the start of the conversation

> My original intent was to discuss step buttons in UnitControl only

**Comment by mirka** [comment]
I updated this PR so it always hides the spin button in a UnitControl. Out of the [paths we can take](https://github.com/WordPress/gutenberg/pull/43985#discussion_r966075889), this is the simpler solution in anticipation of more ergonomic auxiliary controls being added in the near future.

File: packages/components/src/box-control/unit-control.js (hunk 1 above)
Comment (mtias) [reply packages/components/src/box-control/unit-control.js:38]: Ah, I hear you know. Indeed, the idea is the + / – makes sense for unit-less controls. UnitControl should just work with the slider. I cannot think of a case where we'd have the + / – and unit right now.

**Review by ciampo — COMMENTED** [review]
Apart from some snapshots that need updating, code changes LGTM!

I'll defer to @jasmussen or @mtias for final approval on the "strategy"

**Comment by jasmussen** [comment]
Seems fine to me! The slider and plus/minus buttons are the ergonomic replacements, but I wouldn't worry too much about the interim.

**Review by ciampo — APPROVED** [review]
